verkle-trie/src/database/meta.rs

```diff
@@ -123,35 +125,41 @@ impl BranchMeta {
             hash_commitment: Fr::zero(),
         }
     }
-    pub fn from_bytes(bytes: &[u8]) -> BranchMeta {
-        let len = bytes.len();
-        assert_eq!(
-            len,
-            32 + 64,
-            "BranchMeta was not serialised properly, got {}",
-            hex::encode(bytes)
-        );
+}
+
+use crate::{
+    errors::VerkleError,
+    from_to_bytes::{FromBytes, ToBytes},
+};
 
-        use ark_serialize::CanonicalDeserialize;
+impl FromBytes<[u8; 96]> for BranchMeta {
+    fn from_bytes(bytes: [u8; 96]) -> Result<BranchMeta, SerializationError> {
+        let len = bytes.len();
+        if !len == 32 + 64 {
```

Comment:
Same here again, since bytes is not a slice

Reply:
Wrong branch 😨 , will fix this, some issues required this to go back to just a slice when I was working with it